File: solutions/cpp/nth-prime/3/nth_prime.cpp
#include "nth_prime.h"
#include <stdexcept>
#include <vector>
#include <algorithm>

using namespace std;

namespace nth_prime {

int nth(unsigned int n){
    if(n<1){
        throw std::domain_error("Number must be >=1.");
    } 
    vector<int>  primes={2};
    int candidate = 3;
    while(primes.size()<n){ 
            if(none_of(primes.begin(), primes.end(),
                [candidate](int p){return candidate % p  == 0;})){
                    primes.push_back(candidate);    
                }
           candidate += 2;
        }        
        return primes.back();
    }


}  // namespace nth_prime

File: solutions/clojure/nth-prime/3/src/nth_prime.clj
(ns nth-prime)

(defn is-composite? [n primes] 
    (some? (some #(zero? (mod n %)) primes)))

(defn get-primes [n]
  "Returns the first n primes"          
  (loop [next 3
         primes [2]]
    (if (= (count primes) n) primes
      (if (is-composite? next primes)
        (recur (+ next 2) primes)
        (recur (+ next 2) (conj primes next))))))

(defn nth-prime 
  "Returns the prime number in the nth position."
  [n]  
  (if (pos? n)
    (last (get-primes n))  
    (throw (IllegalArgumentException. "n must be >= 1."))))
